### PR TITLE
Chore/upgrade prettier v3

### PR DESCRIPTION
**Pull request by carmanchris31**
WIP of upgrading to prettier v3

There are currently 153 failing tests, most of which are because of thrown errors. Any help with troubleshooting these errors is welcome and appreciated.

Current output of `npm test`:
[test-results.txt](https://github.com/sveltejs/prettier-plugin-svelte/files/11360590/test-results.txt)

### CODE REVIEW

**Comment by dummdidumm** [comment]
@carmanchris31 now that Prettier version 3 is out officially, do you want to continue working on this PR or should I pick it up?

File: package.json [hunk 1]
```diff
@@ -28,13 +28,11 @@
...
     "homepage": "https://github.com/sveltejs/prettier-plugin-svelte#readme",
     "devDependencies": {
-        "@prettier/plugin-pug": "^1.16.0",
         "@rollup/plugin-commonjs": "14.0.0",
         "@rollup/plugin-node-resolve": "11.0.1",
         "@types/node": "^10.12.18",
-        "@types/prettier": "^2.4.1",
         "ava": "3.15.0",
-        "prettier": "^2.7.1",
+        "prettier": "^3.0.0-alpha.11",
```
Comment (benmccann) [inline package.json:35]: this can be updated to `^3.0.0` now

**Comment by carmanchris31** [comment]
> @carmanchris31 now that Prettier version 3 is out officially, do you want to continue working on this PR or should I pick it up?

Please do, thanks! 🙏

**Comment by sebastinez** [comment]
Follow up: Ok just checked on the previous behavior and it seems that the top level nodes aren't recognized as such here, the `_isRoot` property is missing.

https://github.com/sveltejs/prettier-plugin-svelte/blob/85308631451ceb839e43b65218683cf265d73627/src/print/index.ts#L97

So we don't get to extract the prettier:content attributes, and fail later

**Comment by carmanchris31** [comment]
> Follow up: Ok just checked on the previous behavior and it seems that the top level nodes aren't recognized as such here, due to the `_isRoot` property missing.
> 
> https://github.com/sveltejs/prettier-plugin-svelte/blob/85308631451ceb839e43b65218683cf265d73627/src/print/index.ts#L97
> 
> So we don't get to extract the prettier:content attributes, and fail later

Interesting. It sounds like that could be responsible for a vast number of the failures!

**Comment by dummdidumm** [comment]
Found the reason for all the breakage - `embed` call behavior has changed: https://github.com/prettier/prettier/issues/15093 . This is gonna take some refactoring..

**Comment by developomp** [comment]
finally! :partying_face:

**Pull request merged**